mmengine/fileio/backends/oss_backend.py

```diff
@@ -0,0 +1,809 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+import os.path as osp
+import re
+import tempfile
+from contextlib import contextmanager
+from pathlib import Path
+from shutil import SameFileError
+from typing import Generator, Iterator, Optional, Tuple, Union
+
+import mmengine
+from .base import BaseStorageBackend
+
+
+class OSSBackend(BaseStorageBackend):
+    """OSS storage backend for reading (writing) from (to) OSS.
+
+    OSSBackend supports reading and writing data to multiple buckets
+    from Endpoint. If the file path contains the bucket name,
+    OSSBackend will read data from the specified buckets
+    or write data to them. Otherwise, OSSBackend will occur an error.
+
+    Please refer to https://help.aliyun.com/document_detail/85288.html
+    to install the required packages of OSSBackend.
+
+    Args:
+        access_key_id (str, optional): AccessKey ID of your OSS.
+        access_key_secret (str, optional): AccessKey Secret of your OSS.
+        path_mapping (dict, optional): Path mapping dict from local path to
+            ``filepath`` will be replaced by ``dst``. Defaults to None.
+
+    Examples:
+        >>> backend = OSSBackend()
+        >>> filepath1 = 'oss://endpoint/bucket/file'
+        >>> backend.get(filepath1)  # get data from default cluster
+    """
+
+    def __init__(self,
+                 access_key_id: Optional[str] = None,
+                 access_key_secret: Optional[str] = None,
+                 path_mapping: Optional[dict] = None):
+        try:
+            import oss2
+            self.oss2 = oss2
+        except ImportError:
+            raise ImportError('Please install OSS_client to enable '
+                              'OSSBackend.')
+        self._auth = self.oss2.Auth(
+            access_key_id=access_key_id, access_key_secret=access_key_secret)
```

Comment:
Maybe we can also support getting them from environments if they are not specified.

```python
access_key_id = access_key_id or os.getenv('MMENGINE_OSS_ACCESS_KEY_ID')
if access_key_id is None:
    raise ValueError('access_key_id should be specified)
  
access_key_secret = access_key_secret or os.getenv('MMENGINE_OSS_ACCESS_KEY_SECRET', None)
if access_key_secret is None:
    raise ValueError('access_key_secret should be specified)
```